Tests/Realm.Tests/Helpers/RequiresSynchronizationContextAttribute.cs

```diff
@@ -0,0 +1,64 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Realms.Tests
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    internal class RequiresSynchronizationContextAttribute : NUnitAttribute, IWrapTestMethod
+    {
+        public TestCommand Wrap(TestCommand command)
+        {
+            return new ExecuteInSynchronizationContextCommand(command);
+        }
+
+        private class ExecuteInSynchronizationContextCommand : DelegatingTestCommand
+        {
+            private static readonly Type NUnitSynchronizationContext =
+                typeof(NUnitAttribute).Assembly.GetType("NUnit.Framework.Internal.SingleThreadedTestSynchronizationContext");
+
+            public ExecuteInSynchronizationContextCommand(TestCommand innerCommand) : base(innerCommand)
+            {
+            }
+
+            public override TestResult Execute(TestExecutionContext context)
+            {
+                var syncContext = (SynchronizationContext)Activator.CreateInstance(NUnitSynchronizationContext, TimeSpan.FromSeconds(10));
```

Comment:
Should the timespan we pass to the ctor be inferred from the test timeout attribute?

Reply:
This timespan is not configurable - it's [hardcoded in NUnit internally](https://github.com/nunit/nunit/blob/b34eba3ac1aa6957157857bddd116256c634afab/src/NUnitFramework/framework/Internal/AsyncToSyncAdapter.cs#L50) and I'm not sure what other values we should be applying.